FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift

```
@@ -57,14 +78,30 @@ class NeonFrameView: UIView {
     /// - Parameters:
     ///   - titleView: Name of view to be performed
     ///   - containerView: View that will compose the main structure
-    init(titleView: String, containerView: UIView) {
+    init(titleView: String?, containerView: UIView, withCloseButton: Bool = true, withTitle: Bool = true) {
         self.viewTitle = titleView
-        super.init(frame: CGRect.zero)
         self.scrollableView = containerView
+        self.withCloseButton = withCloseButton
+        self.withTitle = withTitle
+        self.scrollableView.translatesAutoresizingMaskIntoConstraints = false
+        super.init(frame: CGRect.zero)
         setupView()
     }
     
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+    
+    override func didMoveToSuperview() {
+        guard let parent = superview else {
+            fatalError("Superview was nil")
```

Do you need to throw a fatal error? Does you always need to have a superview?

---

done